### PR TITLE
Fixed: Can't see private bundle 

### PR DESCRIPTION
**Pull request by ChengshuLi**
Let me know if this is the correct behavior. I actually haven't seen any private bundle before I broke the code. Sorry for the delay.

@percyliang @klopyrev

### CODE REVIEW

**Comment by percyliang** [comment]
I think this is catching the error too deep down.  In general, you want to preserve the error as much as possible (know that it's a PermissionsError rather than 'None', which might because the file doesn't exist).  Can you move the error handling more towards the client side?

**Comment by ChengshuLi** [comment]
@percyliang Feel free to have a look.

File: codalab/rest/legacy.py [hunk 1]
```diff
@@ -203,11 +207,14 @@ def get_summary(info, name):
...
-            for item in info['contents']:
-                item['size_str'] = formatting.size_str(item['size'])
-        return info
+        try:
+            info = self.client.get_target_info(target, 1)
+            if info is not None and info['type'] == 'directory':
+                for item in info['contents']:
+                    item['size_str'] = formatting.size_str(item['size'])
+            return info
+        except PermissionError, e:
```
Comment (percyliang) [inline codalab/rest/legacy.py]: 'try' not necessary here because you're just throwing back the exception...

File: codalab/rest/legacy.py [hunk 2]
```diff
@@ -446,7 +453,10 @@ def post_worksheet_content(uuid):
 @get('/api/bundles/content/<uuid:re:%s>/<path:path>/' % spec_util.UUID_STR)
 def get_bundle_content(uuid, path=''):
     service = BundleService()
-    info = service.get_top_level_contents((uuid, path))
+    try:
```
Comment (percyliang) [inline codalab/rest/legacy.py]: try/except needed?

File: codalab/rest/legacy.py [hunk 3]
```diff
@@ -471,7 +481,11 @@ def get_bundle_info(uuid):
     bundle_info = service.get_bundle_info(uuid)
     if bundle_info is None:
         abort(httplib.NOT_FOUND, 'The bundle is not available')
-    bundle_info.update(service.get_bundle_file_contents(uuid))
+    try:
```
Comment (percyliang) [inline codalab/rest/legacy.py]: try/except needed?

**Comment by klopyrev** [comment]
There's a much better and easier to understand way to make this change:
1) Revert all the current changes.
2) Check whether the bundle is private and don't make the call to get_top_level_contents if it is. To check, simplify check: bundle_info['bundle_type'] != 'private'

If not that, then I agree with Percy: You only need the try / except in a single place: around the get_top_level_contents call. Adding it to all the other places just makes the code confusing as hell.

**Comment by ChengshuLi** [comment]
@percyliang  @klopyrev 

Followed Konstantin's comment

File: codalab/rest/legacy.py [hunk 4]
```diff
@@ -146,8 +146,9 @@ def decode_lines(interpreted):
                 elif isinstance(item['bundle_info'], dict):
                     infos = [item['bundle_info']]
                 for bundle_info in infos:
-                    target_info = self.get_top_level_contents((bundle_info['uuid'], ''))
-                    bundle_info['target_info'] = target_info
+                    if bundle_info['bundle_type'] != 'private':
```
Comment (percyliang) [inline codalab/rest/legacy.py]: Use PrivateBundle.BUNDLE_TYPE

**Comment by ChengshuLi** [comment]
@percyliang

**Comment by ChengshuLi** [comment]
@percyliang

**Comment by ChengshuLi** [comment]
Also, how about private worksheet? Should it behave the same? Now it gives a red alert message if you try to access a private worksheet.

**Comment by percyliang** [comment]
That's fine.  If we want to do that, that should be a different PR.

**Comment by percyliang** [comment]
LGTM

**Pull request merged**